test/unit_tests/cloud/publish_vitals.cpp

```diff
@@ -0,0 +1,247 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <limits>
+
+#define CATCH_CONFIG_MAIN
+#include <catch2/catch.hpp>
+#include <fakeit.hpp>
+
+#include "system_publish_vitals.h"
+#include "../test/unit_tests/mock_types.h"
+
+// ASSUMPTION!!! - Period "getter" method works correctly - without being testing.
+
+TEST_CASE("Initialized values", "[VitalsPublisher::VitalsPublisher]")
+{
+    particle::mock_type::Timer t;
+
+    SECTION("When constructed, then period is initialized to MAX value.")
+    {
+        // Arrange
+
+        // Act
+        particle::system::VitalsPublisher<particle::mock_type::Timer> vp(nullptr, &t);
+
+        // Assert
+        CHECK(std::numeric_limits<system_tick_t>::max() == vp.period());
+    }
+}
+
+TEST_CASE("Cleanup", "[VitalsPublisher::~VitalsPublisher]")
+{
+    fakeit::Mock<particle::mock_type::Timer> mock_timer;
+
+    // Configure Mock
+    fakeit::When(Method(mock_timer, changePeriod)).AlwaysReturn(true);
+    fakeit::When(Method(mock_timer, dispose)).AlwaysReturn();
+    fakeit::When(Method(mock_timer, isActive)).AlwaysReturn(false);
+    fakeit::When(Method(mock_timer, reset)).AlwaysReturn();
+    fakeit::When(Method(mock_timer, start)).AlwaysReturn();
+    fakeit::When(Method(mock_timer, stop)).AlwaysReturn();
+
+    SECTION("When destroyed, then the timer will be disposed.")
+    {
+        // Arrange
+
+        // Act
+        {
+            particle::system::VitalsPublisher<particle::mock_type::Timer> vp(nullptr,
+                                                                             &mock_timer.get());
+        }
+
+        // Assert
+        fakeit::Verify(Method(mock_timer, dispose));
+    }
+}
+
+TEST_CASE("Stop publishing", "[VitalsPublisher::disablePeriodicPublish]")
+{
+    fakeit::Mock<particle::mock_type::Timer> mock_timer;
+    particle::system::VitalsPublisher<particle::mock_type::Timer> vp(nullptr, &mock_timer.get());
+
+    // Configure Mock
+    fakeit::When(Method(mock_timer, changePeriod)).AlwaysReturn(true);
+    fakeit::When(Method(mock_timer, dispose)).AlwaysReturn();
+    fakeit::When(Method(mock_timer, isActive)).AlwaysReturn(false);
+    fakeit::When(Method(mock_timer, reset)).AlwaysReturn();
+    fakeit::When(Method(mock_timer, start)).AlwaysReturn();
+    fakeit::When(Method(mock_timer, stop)).AlwaysReturn();
+
+    SECTION("When disabled, then the timer will be disposed.")
+    {
+        // Arrange
```

Comment:
The comments separating the setup/execute/validate sections are good,  although Catch supports Given, When, Then blocks to make this explicit in the code, with the added benefit that these items can be nested for reuse. (Often individual tests have the common setup or execution steps.)  Here's an example: https://github.com/particle-iot/device-os/pull/1732/files#diff-deee133a44ce35e2149c0ac5feddc545R26